src/bigint.cpp

```diff
@@ -230,7 +235,201 @@ Bigint &Bigint::operator*=(int const &b)
+              window_size++;
+              tmpx0 = to_string(*i);
+              if(i!=p.number.end()-1 && tmpx0.size()<9){
+                  tmpx0 = std::string(9-tmpx0.size(),'0') + tmpx0;
+              }
+              tmpx0 = tmpx0[0];
+              tmpx1 = tmpx1 + tmpx0;
+              c=tmpx1;
+          }
+
+      }
+
+      return c;
+}
+//Division
+
+std::vector<Bigint> Bigint::divide(Bigint q){
+
+    /*Algorithm used is "Double division algorithm"*/
+
+    Bigint p = *this;
+    std::vector<Bigint> answer;
+    int look_up_table_size=4;
+    std::vector<Bigint> look_up(look_up_table_size);
+    bool done_flag=false ;
+    Bigint tmp_quotient, sum_quotient, tmpx1, tmpx2;
+
+    Bigint zero("0");
+
+    if(q==zero || q == 0){
+    //std::cout << "Error: Dividing by zero" << std::endl;
+    //exit (EXIT_FAILURE);
+    throw "Dividing by zero!";
+    }
+
+    bool this_sign = this -> positive;
+    bool q_sign = q.positive;
+
+    p.positive = true;
+    q.positive = true;
+
+
+    look_up[0]=q;
+    look_up[1]=q*2;
+    look_up[2]=q*4;
+    look_up[3]=q*8;
+
+
+    while(true){
+
+       Bigint sub_p=sub_number(p, q);  // cut a portion from the dividened equal in size with the divisor
+
+        for(int i=0;i<look_up_table_size;i++){ // looking in the look_up table
+
+            if(sub_p<look_up[i] && i!=0){
+                tmpx1=look_up[i-1];
+                tmp_quotient = std::pow(2,i-1);
+                break;
+            }
+            else if(sub_p<look_up[i] && i==0){
+
+                if(q.digits()>=p.digits()){
+                    done_flag=true;
+                    break;
+                }
+            }
+            else if(sub_p==look_up[i] || (sub_p>look_up[i] && i==look_up_table_size-1)){
+                 tmpx1=look_up[i];
+                 tmp_quotient= std::pow(2,i);
+                break;
+            }
+        }
+
+        if(done_flag){
+            answer.push_back(sum_quotient);
+            answer.push_back(sub_p);
+            break;
+        }
+
+        std::string temppp(p.digits()-(sub_p.digits()),'0');
+        temppp="1"+temppp;
+        tmpx2=temppp;
+
+        if(sum_quotient.digits()==0)
+            sum_quotient=tmp_quotient*tmpx2;
+        else
+            sum_quotient=sum_quotient+(tmp_quotient*tmpx2);
+
+        tmpx1=tmpx1*tmpx2;
+        p=p-tmpx1;
+
+    }
+
+    if(this_sign == false && q_sign == false){
+        answer[0].positive = true;
+    }
+
+    else if(this_sign == false || q_sign == false){
+        if( this -> is_even() && q.is_even())
+            answer[0].positive = false;
+        else{
+            answer[0] = answer[0] + Bigint("1");
+            answer[0].positive = false;
+        }
+    }
+
+    else{
+        answer[0].positive = true;
+    }
+
+    return answer;
+
+}
+
+std::vector<Bigint> Bigint::operator/(Bigint q){
+    try{
+        return divide(q);
+    }
+    catch(const char* msg){
+        std::cerr << msg << std::endl;
+        std::terminate();
```

Comment:
no, user won't catch it. It is caught in the division function interface only and printed on screen as shown above.
It is a little tricky and i don't know how to handle the exception, so please can you clarify more.